src/app/Recordings/ReportFrame.tsx

```diff
@@ -57,23 +59,24 @@ export const ReportFrame: React.FunctionComponent<ReportFrameProps> = React.memo
     if (!props.isExpanded) {
       return;
     }
-    const sub = context.reports
-      .report(recording)
-      .pipe(first())
-      .subscribe(
-        (report) => setReport(report),
-        (err) => {
-          if (isGenerationError(err)) {
-            err.messageDetail.pipe(first()).subscribe((detail) => setReport(detail));
-          } else if (isHttpError(err)) {
-            setReport(err.message);
-          } else {
-            setReport(JSON.stringify(err));
+    addSubscription(
+      context.reports
+        .report(recording)
+        .pipe(first())
+        .subscribe(
+          (report) => setReport(report),
+          (err) => {
+            if (isGenerationError(err)) {
+              err.messageDetail.pipe(first()).subscribe((detail) => setReport(detail));
+            } else if (isHttpError(err)) {
+              setReport(err.message);
+            } else {
+              setReport(JSON.stringify(err));
+            }
           }
-        }
-      );
-    return () => sub.unsubscribe();
-  }, [context, context.reports, recording, isExpanded, setReport, props, props.isExpanded, props.recording]);
+        )
+    );
+  }, [addSubscription, context.reports, recording, isExpanded, setReport, props, props.isExpanded, props.recording]);
```

Comment:
I think we can remove `props.isExpanded` and `props.recording` from this list as they are the same as `recording` and `isExpanded`.